src/main/java/com/srans/nestserver/controller/HostelController.java

```diff
@@ -17,59 +22,271 @@
 import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
 
 import com.srans.nestserver.exception.ResourceNotFoundException;
+import com.srans.nestserver.model.Floor;
 import com.srans.nestserver.model.Hostel;
+import com.srans.nestserver.model.Room;
+import com.srans.nestserver.repository.BedRepository;
+import com.srans.nestserver.repository.FloorRepository;
 import com.srans.nestserver.repository.HostelRepository;
+import com.srans.nestserver.repository.RoomRepository;
+import com.srans.nestserver.service.StorageService;
+import com.srans.nestserver.util.NSException;
 
-@CrossOrigin(origins = "http://localhost:4200",allowedHeaders = "*")
+@CrossOrigin(origins = "http://localhost:4200", allowedHeaders = "*")
+@RequestMapping("api/v1")
 @RestController
-@RequestMapping("/api/v1")
 public class HostelController {
+
+	private Logger logger = LoggerFactory.getLogger(HostelController.class);
+
 	@Autowired
 	private HostelRepository hostelRepository;
 
-	@GetMapping("/hostels")
-	public List<Hostel> getAllHostels() {
-		return hostelRepository.findAll();
+	@Autowired
+	private FloorRepository floorRepository;
+
+	@Autowired
+	private RoomRepository roomRepository;
+
+	@Autowired
+	private BedRepository bedRepository;
+
+	@Autowired
+	private StorageService storageService;
+
+	/*
+	 * @GetMapping("/hostels") public List<Hostel> getAllPosts() { return
+	 * hostelRepository.findAll(); }
+	 */
+
+	@PostMapping("/hostels")
+	public Hostel saveHostel(@Valid @RequestBody Hostel hostel) throws NSException {
+
+		logger.info("IN::POST::/hostels::saveHostel::" + hostel);
+
+		Hostel responseHostel = hostelRepository.save(hostel);
+
+		// SAVE Database stuff here
+
+		responseHostel.getfloors().forEach(floor -> {
```

Comment:
SaveHostel , looks good as you saved all sub equivalent associated objects been saved, i dont see same structure of implementation followed on other /hostels related API calls, this should be consistent across all